lib/imagery/sigfile.c

```diff
@@ -17,56 +17,42 @@
 /*!
    \brief Create signature file
 
-   \param group group name
-   \param subgroup subgroup name in given group
    \param name signature filename
 
-   \return pointer to FILE*
+   \return pointer to FILE
    \return NULL on error
 */
-FILE *I_fopen_signature_file_new(const char *group,
-				 const char *subgroup, const char *name)
+FILE *I_fopen_signature_file_new(const char *name)
 {
-    char element[GPATH_MAX];
-    char group_name[GNAME_MAX], group_mapset[GMAPSET_MAX];
     FILE *fd;
 
-    if (!G_name_is_fully_qualified(group, group_name, group_mapset)) {
-	strcpy(group_name, group);
-    }
+    /* create sig directory */
+    G__make_mapset_element_misc("signatures", "sig");
 
-    /* create sigset directory */
-    sprintf(element, "%s/subgroup/%s/sig", group_name, subgroup);
-    G__make_mapset_element_misc("group", element);
-
-    sprintf(element, "subgroup/%s/sig/%s", subgroup, name);
-
-    fd = G_fopen_new_misc("group", element, group_name);
+    fd = G_fopen_new_misc("signatures", name, "sig");
```

Comment:
You are rightfully confused. Functions dealing with mapset elements are confusing. Their documentation is outdated (G_fopen_new_misc describes two of three arguments!). 
G_fopen_new_misc(dir, element, name) is a great example. Test for yourself:
```
from grass.lib.gis import G_fopen_new_misc
G_fopen_new_misc("dir", "element", "name")
```
And the outcome is (drum roll): MAPSET/dir/name/element
Thus "signatures" = directory, "name" = element, "sig" = name to form easy to use MAPSET/signatures/sig/name

Reply:
Right and I think your usage goes against intended use of this function, that's why you have element=name and name=sig. I'm more and more convinced that the intention there is specifically raster map cell_misc directory: dir=cell_misc, element=range, and name=elevation. The order of parameters may be either result of trying to fit this with the original `cell/elevation` layout or result of `object_type_name > object_name` thinking where object_type_name=element and object_name=name and all these go into some dir.

I don't think you should use these functions as "create many nested directories" as there are different constraints for different directories. That's the whole motivation for new functions in #1681. For some directories, race condition during creation is perfectly fine (signatures, sig, cell_misc) and for some it is not (elevation subdir in cell_misc).

`signatures/sig` seems to me a new thing in the sense that it is a type (signatures) with a subtype (sig), so I think you need new functions in the library for that.

To bring another perspective here, it seems to me that G_fopen_new_misc is basically what should be used for all raster files in the future. Just instead of cell_misc, it will use `raster` and all of fcell, cellhr, ... would be files inside a sub-directory of raster named according to the raster map name, creating raster/elevation/fcell hiearchy. Your use would than be "signatures" = raster (so far so good), but then "sig" = elevation and "name" = fcell.